### PR TITLE
Structured logging for essential log lines

### PR DESCRIPTION
**Pull request by sr**
Use structured logging ([logfmt](https://brandur.org/logfmt) formatted) for a few essential log lines. Example output:

```
at=start type=daily
at=start type=daily dir=/home/sr
at=finish type=daily dir=/home/sr duration=23s
at=start type=daily dir=/home/cbk/data/fastmail-carddav
at=finish type=daily dir=/home/cbk/data/fastmail-carddav duration=13s
at=start type=daily dir=/home/cbk/data/fastmail-imap
at=finish type=daily dir=/home/cbk/data/fastmail-imap duration=143s
at=start type=daily dir=/home/cbk/data/github
at=finish type=daily dir=/home/cbk/data/github duration=59s
at=start type=daily dir=/home/cbk/data/irclogs
at=finish type=daily dir=/home/cbk/data/irclogs duration=19s
at=start type=daily dir=/home/cbk/data/pinboard
at=finish type=daily dir=/home/cbk/data/pinboard duration=12s
at=start type=daily dir=/home/cbk/data/soundcloud-likes
at=finish type=daily dir=/home/cbk/data/soundcloud-likes duration=642s
at=finish type=daily duration=943s
```

This makes it easier to search logs with tools such as [Papertrail](https://papertrail.com) and [Splunk](https://splunk.com) or e.g. create alerts for `at=error`.

### CODE REVIEW

File: acts [hunk 1]
```diff
@@ -149,7 +149,7 @@ archives=$($tarsnap --list-archives | sort -n)
 # PART 4: Trim old backups.
 
 if [ "$backuprc" != "0" ] ; then
-    log_message "One of the backups failed -- not deleting old backups"
+    log_verbose "at=error type=$archivetype"
```
Comment (alexjurkiewicz) [inline acts]: I'm wondering if `error` is the right level for this line, it is mostly added to aid people read logfiles (since it just ensures any error is referred to be two lines). What do you think?

Comment (alexjurkiewicz) [inline acts]: Actually, I also feel this modification removes too much context -- the user would have to read `acts` source to understand what is happening.

**Comment by alexjurkiewicz** [comment]
The format feels a bit ruby-ish ("`at=`" ugh) but I like the principle. Based on the blog post you linked you can add a "positional argument" (for want of a better term), I think that would be helpful in aiding readability. Eg:

```
at=start start-acts type=daily
at=start backup-directory type=daily dir=/home/sr
at=finish backup-directory type=daily dir=/home/sr duration=23s
at=error backup-failed-not-deleting-old-backups type=daily
at=finish finish-acts type=daily duration=943s
```

File: acts (hunk 1 above)
Comment (sr) [inline acts]: Yeah agreed. Just going to revert this change. The same information is already logged with more context above:

https://github.com/alexjurkiewicz/acts/pull/8/files#diff-fa1679bbd1d49c770b4ff7d0074bc24eR139

File: acts [hunk 2]
```diff
@@ -114,7 +114,7 @@ else
     # There's no yearly backup
     archivetype="yearly"
 fi
-log_verbose "Creating $archivetype backup"
+log_verbose "at=start type=$archivetype"
```
Comment (sr) [inline acts]: Reverting this as well. It's somewhat redundant with log messages below that have more context.

**Comment by alexjurkiewicz** [comment]
Thanks for this PR, finally merged! I ended up converting all output to logfmt style, for consistency. Would you mind sanity checking the messages for me? Let me know if any don't make sense.

**Comment by sr** [comment]
Looks great, thanks!
